Atomic.md

```diff
@@ -0,0 +1,278 @@
+Atomic placeholder operations in VFSForGit on MacOS
+===================================================
+
+A potential hole in our current design is in our creation of
+placeholders. Today, when creating a placeholder directory, at a high
+level, we go through the following workflow:
+
+-   Create placeholder folder/file in the virtualization root
```

Comment:
> placeholder folder/file

Nit: Everywhere else you use the term "directory" rather than "folder".  Consider doing the same here for consistency.